src/map/addPopUp.js

```diff
@@ -3,174 +3,181 @@ import { getCurrentState } from '../store/actions/actions';
+                                            datum[k] === '')) {
+                                        delete datum[k];
+                                    }
+                                });
+                            } else {
+                                const found = [];
+                                const rxp = /{{([^}]+)}/g;
+                                const str = layer.labels ? layer.labels.label : null;
+                                for (let c = rxp.exec(str); c !== null; c = rxp.exec(str)) {
+                                    found.push(c[1]);
+                                }
+                                // while (curMatch = rxp.exec(str)) {
+                                //   found.push(curMatch[1]);
+                                // }
+                                found.forEach(i => {
+                                    rowItem[`${i}`] = rowItem[`${i}`].toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',');
+                                });
+                            }
+                            // Add header and body to popup with data from layer
+                            if (rowItem[layer.popup.header]) {
+                                // for null values
+                                if (layer.popup && layer.popup.hideNulls) {
+
+                                    content =
+                                        `<div><b>${rowItem[layer.popup.header]}</b></div>` +
+                                        `<div><center>${Mustache.render(layer.popup.body, commaFormatting(layer, datum, true))}</center></div>`;
+                                } else {
+                                    const bodyProperties = layer.popup.body.match(/{{(.*?)\}}/g).map(val => val.replace(/{{?/g, '').replace(/}}?/g, ''));
+                                    bodyProperties.forEach(val => {
+                                        // Check if rowItem[val] is a string and if it has ,
+                                        const contentArr = typeof rowItem[val.trim()] !== 'number' &&
+                                            rowItem[val.trim()].includes(',') ? rowItem[val.trim()].split(',') : [];
+                                        if (contentArr.length > 1) {
+                                            rowItem[val] = contentArr.join(', ');
+                                        }
+                                    });
+                                    content =
+                                        `<div>` +
+                                        `<div><b>${row[layer.popup.header]}</b></div>` +
+                                        `<div><center>${Mustache.render(
+                                            layer.popup.body,
+                                            commaFormatting(layer, rowItem, true)
+                                        )}</center></div>` +
+                                        `</div>`;
+                                }
```

Comment:
Why do we need two blocks here? Why do we not do check for `,` if we're hiding nulls?

Reply:
I will need to refactor this code